accounts.go

```diff
@@ -342,3 +347,58 @@ func applyUpdateMask(mask *field_mask.FieldMask, msg protoreflect.ProtoMessage,
 	fmutils.Filter(msg, allowedFields)
 	return nil
 }
+
+func (srv *accountsAPI) AuthenticateGoogle(ctx context.Context, in *accountsv1.AuthenticateGoogleRequest) (*accountsv1.AuthenticateGoogleResponse, error) {
+
+	code := in.Code
+
+	token, err := srv.googleOAuth.Exchange(context.Background(), code)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to exchange token: "+err.Error())
+	}
+
+	userinfo, err := srv.googleOAuth.Client(context.Background(), token).Get("https://www.googleapis.com/oauth2/v2/userinfo")
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to get userinfo: "+err.Error())
+	}
+
+	defer userinfo.Body.Close()
+	content, err := ioutil.ReadAll(userinfo.Body)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to read response body: "+err.Error())
+	}
+
+	var userInfo map[string]interface{}
+	err = json.Unmarshal(content, &userInfo)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to unmarshal response body: "+err.Error())
+	}
+	email := userInfo["email"].(string)
+	id := userInfo["id"].(string)
+
+	// check if the user does not exist, register the user
+	_, err = srv.repo.Get(ctx, &models.OneAccountFilter{Email: email})
+	if err != nil {
+		fmt.Printf("Register the user if not exists ...")
+		hashed, err := bcrypt.GenerateFromPassword([]byte(id), 8)
+		if err != nil {
+			srv.logger.Error("bcrypt failed to hash password", zap.Error(err))
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		// get the userinfo name in string format
+		name := userInfo["name"].(string)
+
+		_, err = srv.repo.Create(ctx, &models.AccountPayload{Email: &email, Name: &name, Hash: &hashed})
+		if err != nil {
```

Comment:
When a new user is created we should notify the notes-service to create a workspace associate with the account_id, 
you can look at CreateAccount function for the CreateWorkspace request

Reply:
Awesome, I hadn't thought about that

Reply:
I think you can also create a private method `CreateUser` called in account creation and here to not duplicate the code

Reply:
@gabriel654165, I should create it in the ``AccountsRepository`` model  ?